server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxAggregatorTests.java

```diff
@@ -122,4 +145,110 @@ private void testCase(Query query, CheckedConsumer<RandomIndexWriter, IOExceptio
+                document.add(new StringField("id", Integer.toString(docID), Field.Store.NO));
+                document.add(pointFieldFunc.apply(nextValue));
+                documents.add(document);
+                docID ++;
+            }
+        }
+        // insert some documents without a value for the metric field.
+        for (int i = 0; i < 3; i++) {
+            Document document = new Document();
+            documents.add(document);
+        }
+        indexWriter.addDocuments(documents);
+        Collections.sort(values, Comparator.comparingDouble(t -> t.v2().doubleValue()));
+        try (IndexReader reader = DirectoryReader.open(indexWriter)) {
+            LeafReaderContext ctx = reader.leaves().get(0);
+            Number res = MaxAggregator.findLeafMaxValue(ctx.reader(), "number" , pointConvertFunc);
+            assertThat(res, equalTo(values.get(values.size()-1).v2()));
+        }
+        for (int i = values.size()-1; i > 0; i--) {
+            indexWriter.deleteDocuments(new Term("id", values.get(i).v1().toString()));
+            try (IndexReader reader = DirectoryReader.open(indexWriter)) {
+                LeafReaderContext ctx = reader.leaves().get(0);
+                Number res = MaxAggregator.findLeafMaxValue(ctx.reader(), "number" , pointConvertFunc);
+                if (res != null) {
+                    assertThat(res, equalTo(values.get(i - 1).v2()));
+                } else {
+                    assertAllDeleted(ctx.reader().getLiveDocs(), ctx.reader().getPointValues("number"));
+                }
+            }
+        }
+        indexWriter.deleteDocuments(new Term("id", values.get(0).v1().toString()));
+        try (IndexReader reader = DirectoryReader.open(indexWriter)) {
+            LeafReaderContext ctx = reader.leaves().get(0);
+            Number res = MaxAggregator.findLeafMaxValue(ctx.reader(), "number" , pointConvertFunc);
+            assertThat(res, equalTo(null));
+        }
+        indexWriter.close();
+        directory.close();
+    }
+
+    // checks that documents inside the max leaves are all deleted
+    private void assertAllDeleted(Bits liveDocs, PointValues values) throws IOException {
+        final byte[] maxValue = values.getMaxPackedValue();
+        int numBytes = values.getBytesPerDimension();
+        final boolean[] seen = new boolean[1];
+        values.intersect(new PointValues.IntersectVisitor() {
+            @Override
+            public void visit(int docID) {
+                throw new AssertionError();
+            }
+
+            @Override
+            public void visit(int docID, byte[] packedValue) {
+                assertFalse(liveDocs.get(docID));
+                seen[0] = true;
+            }
+
+            @Override
+            public PointValues.Relation compare(byte[] minPackedValue, byte[] maxPackedValue) {
+                if (FutureArrays.compareUnsigned(maxPackedValue, 0,  numBytes, maxValue, 0, numBytes) == 0) {
```

Comment:
Arrays#equals?

Reply:
Same here

Reply:
I switched to FutureArrays#equals which is equivalent